packages/connections/src/components/connection-list/connection-menu.tsx

```diff
@@ -0,0 +1,127 @@
+import { css } from '@emotion/css';
+import React, { useReducer } from 'react';
+import {
+  IconButton,
+  Icon,
+  Menu,
+  MenuItem,
+  Toast,
+  ToastVariant,
+  spacing,
+} from '@mongodb-js/compass-components';
+
+const dropdownButtonStyles = css({
+  color: 'white',
+  position: 'absolute',
+  right: spacing[1],
+  top: spacing[2],
+  bottom: 0,
+});
+
+const toastStyles = css({
+  button: {
+    position: 'absolute',
+  },
+});
+
+type State = {
+  toastVariant: ToastVariant.Success | ToastVariant.Warning;
+  error: string;
+  toastOpen: boolean;
+};
+
+const defaultToastState: State = {
+  toastOpen: false,
+  toastVariant: ToastVariant.Success,
+  error: '',
+};
+
+type Action =
+  | { type: 'show-success-toast' }
+  | { type: 'show-warning-toast'; error }
+  | { type: 'close-toast' };
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'show-success-toast':
+      return {
+        ...state,
+        toastOpen: true,
+        toastVariant: ToastVariant.Success,
+        error: '',
+      };
+    case 'show-warning-toast':
+      return {
+        ...state,
+        toastOpen: true,
+        toastVariant: ToastVariant.Warning,
+        error: action.error,
+      };
+    case 'close-toast':
+      return {
+        ...state,
+        toastOpen: false,
+      };
+    default:
+      return state;
+  }
+}
+
+function ConnectionMenu({
+  connectionString,
+}: {
+  connectionString: string;
+}): React.ReactElement {
+  const [{ error, toastOpen, toastVariant }, dispatch] = useReducer(reducer, {
+    ...defaultToastState,
+  });
+
+  async function copyConnectionString(connectionString: string) {
+    try {
+      await navigator.clipboard.writeText(connectionString);
+      dispatch({
+        type: 'show-success-toast',
+      });
+    } catch (err) {
+      dispatch({
+        type: 'show-warning-toast',
+        error: err,
+      });
+    }
+  }
+
+  return (
+    <>
+      <Toast
+        className={toastStyles}
+        variant={toastVariant}
+        title={toastVariant === ToastVariant.Success ? 'Success!' : 'Error'}
+        body={
+          toastVariant === ToastVariant.Success
+            ? 'Copied to clipboard.'
+            : `${error}`
+        }
+        open={toastOpen}
+        close={() => dispatch({ type: 'close-toast' })}
+      />
```

Comment:
These toasts are popping up and overlaying when you copy different urls, on top of that because the leafygreen component inserts the container for the toast immediately when component is rendered on the screen the order of toasts in the DOM (the one you would see on the top) doesn't correlate with the latest url you copied. This is definitely a corner case here, but if we are planning to use toasts in any other parts of the app I think we need a better strategy for how we render these. Should we open a ticket for that?

Reply:
Added toast ticket here: https://jira.mongodb.org/browse/COMPASS-5259 to avoid multiple toasts in the dom.